src/docs/arch-column-row.dox

```diff
@@ -0,0 +1,99 @@
+/*! @arch_page arch-column-row Row Store & Column Store
+
+@section column_row_definition Definition
+
+A row oriented storage is an access method where all columns of a row are stored
+together, the data is stored by row, while a column oriented system stores its data tables by
+column. WiredTiger supports both and each management system is represented internally by a certain
+type of B-Tree data structure.
+
+Keys and values in row-stores can be of variable length, thus, row-stores implement a generalized
+version of a B-Tree. Furthermore, keys are explicitly stored. Column-stores use a 64-bit unsigned
+integer record id as keys, thus, column-stores implement a specialized version of a B-Tree where the
```

Comment:
I'm still a little unhappy with `generalized` and `specialized`, here. I don't know what they mean and I don't think they really add anything: there are a million Btree variants.

Reply:
I have removed those two words in 924b8684ad0ca9fbbd89cf126d51501c3809cb26